compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -695,9 +696,16 @@ class ExpressionCodegen(
         val elseLabel = Label()
         val thenBranch = branch.result
         //TODO don't generate condition for else branch - java verifier fails with empty stack
-        val elseBranch = branch is IrElseBranch
-        if (!elseBranch) {
+        // Else branch comes from "if (ALWAYS_TRUE)", where ALWAYS_TRUE can be handwritten, inferred by compiler,
+        // or a construct of IrElseBranch.
+        val isElseBranch = branch.condition.isTrueConst()
```

Comment:
@ting-yuan Please name variable properly: 'true' condition could also be in non-else